### PR TITLE
EXPORT/RELEASE changes for 7.3-preview

### PR DESCRIPTION
**Pull request by herveyw-msft**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [x] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/keys.json [hunk 1]
```diff
@@ -1607,6 +1607,21 @@
...
           "$ref": "#/definitions/JsonWebKey",
           "description": "The export key encryption Json web key. This key MUST be a RSA key that supports encryption."
+        },
+        "enc": {
+          "x-ms-client-name": "enc",
+          "type": "string",
+          "minLength": 1,
+          "description": "The encryption algorithm to use to protected the exported key material",
+          "enum": [
+            "CKM_RSA_AES_KEY_WRAP",
```
Comment (heaths) [inline specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/keys.json:1617]: These names would not be allowed in the SDK, nor are they consistent with existing enum value names in the swagger (many of which are already non-standard names like "A192CBCAD"). Could these be made more friendly like the other enums? For example, each value can also have an `x-ms-enum` name that is maybe something like `chmRsaAesKeyWrap` which would then be consistent across languages.

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/keys.json [hunk 2]
```diff
@@ -1607,6 +1607,21 @@
         "kek": {
           "$ref": "#/definitions/JsonWebKey",
           "description": "The export key encryption Json web key. This key MUST be a RSA key that supports encryption."
+        },
+        "enc": {
+          "x-ms-client-name": "enc",
```
[the remark this answers is not in the document]
Comment (heaths) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/keys.json:1612]: Same comment applies below.

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/keys.json (hunk 1 above)
Comment (heaths) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/keys.json:1617]: Same comment applies below.

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/keys.json [hunk 3]
```diff
@@ -1607,6 +1607,21 @@
...
+          "type": "string",
+          "minLength": 1,
```
Comment (heaths) [inline specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/keys.json:1614]: If 1 value is required, should `enc` be a required property, as indicated in the `required` JSON schema array? If the property is required and 1 value is required, both should be authored.

Comment (heaths) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/keys.json:1614]: Same comment applies below.

**Review by heaths — CHANGES_REQUESTED** [review]

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/keys.json (hunk 1 above)
Comment (herveyw-msft) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/keys.json:1617]: The other enums do not have "friendly" values, they are the values from JSON Web Algorithms. Some have additional descriptions, others do not. The particular values use, or are derived from the mechanism names in PKCS#11.

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/keys.json (hunk 3 above)
Comment (herveyw-msft) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/keys.json:1614]: minLength is 1, I don't think that is the same as "required". The definition here matches other enum definitions in this file.

**Comment by herveyw-msft** [comment]
Re: Breaking changes. This is an unreleased preview protocol version, by definition there are no breaking changes.

**Comment by heaths** [comment]
@herveyw-msft:
> Re: Breaking changes. This is an unreleased preview protocol version, by definition there are no breaking changes

Just FYI: this will actually be changing because previews are supposed to be supported for 90d, but can be exempted for now. We'll probably have to end up using feature branches, which is what the REST review board and I have been talking about recently. I can provide more details later.

**Review by heaths — APPROVED** [review]

**Comment by heaths** [comment]
@tjprescott can you please force merge?

**Comment by herveyw-msft** [comment]
@heaths 7.3-preview is not implemented on either AKV or AKV MHSM. It has not made it to preview yet, so the 90D clock has not even started.

**Comment by heaths** [comment]
@herveyw-msft we can take that conversation offline, but it does raise an interesting question: if the service hasn't implemented this yet, should this PR be merged yet? If it doesn't but within days, that's probably fine. We can hold off on executing tests against this endpoint until it's ready and just generate/tweak the code for now.

/cc @chlowell @mccoyp

**Review by tjprescott — APPROVED** [review]

**Comment by tjprescott** [comment]
@lmazuel can you merge this? This swagger has not been released in any form, so the breaking change is a false positive.

**Comment by tjprescott** [comment]
@lmazuel ping

**Pull request merged**